File: src/main/java/org/axostudio/axohologram/importer/HologramImporter.java
package org.axostudio.axohologram.importer;

import java.util.Collection;

public interface HologramImporter {

    String id();

    String displayName();

    boolean isAvailable();

    Collection<String> availableHolograms();

    ImportResult importHologram(String name);

    ImportResult importAll();

    default ImportResult importAllWithoutBackup() {
        return importAll();
    }
}

File: src/main/kotlin/org/axostudio/axohologram/feature/importer/HologramImporter.kt
package org.axostudio.axohologram.feature.importer

interface HologramImporter {
    fun id(): String
    fun displayName(): String
    fun isAvailable(): Boolean
    fun availableHolograms(): Collection<String>
    fun importHologram(name: String): ImportResult
    fun importAll(): ImportResult
    fun importAllWithoutBackup(): ImportResult = importAll()
}
